### DISCUSSION TITLE
Usage with Laravel Vapor

### DISCUSSION THREAD

**Discussion by Shnilj** [brefphp/bref#823] [Support]
I am currently using [Laravel Vapor](https://vapor.laravel.com/) to run Serverless PHP through AWS. By default, Vapor has an option to use Imagick based on this repository [see](https://docs.vapor.build/1.0/projects/environments.html#imagick-support). I was wondering how I can extend this functionality to add SQLSrv support to my project.

Thanks in advance!

**Comment by sonnysantino** [comment]
Hola @Shnilj, please look at https://github.com/brefphp/extra-php-extensions, there's a layer for SQLSrv already 🙌

**Comment by liepumartins** [comment]
Apparently I have the same question. Feels like the vapor and bref layers differ somewhat.

@sonnysantino question is not about existence of the layer, but using it with vapor instead of bref.

@Shnilj any luck?

I found this https://flareapp.io/blog/8-installing-blackfire-on-laravel-vapor and tried to do same steps.

1. `git clone https://github.com/brefphp/extra-php-extensions`
2. `cd extra-php-extensions && make layers layer=yaml`
3. Take the necessary zip file from export/ directory
4. Create new layer at AWS lambda, get ARN
5. Update `vapor.yml` as follows:
```yml
...
        layers:
          - vapor:php-8.1:al2
          #- arn:aws:lambda:eu-north-1:209497400698:layer:php-81-fpm:34 # also tried bref runtimes instead of vapor's https://runtimes.bref.sh/
          #- arn:aws:lambda:eu-north-1:209497400698:layer:php-81:34 # however using these breaks function completely
          - arn:aws:lambda:eu-north-1:xxxxxxxxxxxx:layer:yaml-php-81:1 # the custom layer, aws id masked
...
```
6. Create php/conf.d/yaml.ini with `extension=extension=/opt/bref-extra/yaml.so`
7. deploy

PHP info says that 

|Additional .ini files parsed | /opt/vapor/etc/php/conf.d/vapor.ini, /var/task/php/conf.d/yaml.ini |
|-|-|

However `function_exists('yaml_parse')` returns false;


And cloudwatch shows
`PHP Warning:  PHP Startup: Unable to load dynamic library '/opt/bref-extra/yaml.so' (tried: /opt/bref-extra/yaml.so (libyaml-0.so.2: cannot open shared object file: No such file or directory), /opt/bin//opt/bref-extra/yaml.so.so (/opt/bin//opt/bref-extra/yaml.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0`

**Comment by liepumartins** [comment]
Looks like I found a solution.

LD_LIBRARY_PATH | /opt/lib:/opt/lib/bref:/lib64:/usr/lib64:/var/runtime:/var/runtime/lib:/var/task:/var/task/lib
-- | --

Tried to modify at first, but seems it has no effect or `ENV LD_LIBRARY_PATH=/opt/bref/lib64:$LD_LIBRARY_PATH` is incorrect.
`phpinfo()` shows LD_LIBRARY_PATH unaltered.

Anyway, I instructed the Dockerfile to copy system libraries `libyaml-0.so.2` to a place where vapor environment will look for it: `/opt/lib/bref` (the default is `/opt/bref/lib64`)
`COPY --from=ext /opt/bref/lib/libyaml-0.so.2 /opt/bref/lib64/libyaml-0.so.2`
changed to
`COPY --from=ext /opt/bref/lib/libyaml-0.so.2 /opt/lib/bref/libyaml-0.so.2`
